src/applications/personalization/dashboard-2/components/Dashboard.jsx

```diff
@@ -72,45 +72,101 @@ const Dashboard = ({
     ],
   );
 
-  if (showLoader) {
-    return <Loader />;
-  } else {
-    return (
-      <div className="dashboard">
-        {props.showNameTag && (
-          <NameTag
-            showUpdatedNameTag
-            totalDisabilityRating={props.totalDisabilityRating}
-          />
-        )}
-        <div className="vads-l-grid-container medium-screen:vads-u-padding--2 small-desktop-screen:vads-u-padding--0">
-          <Breadcrumbs>
-            <a href="/" key="home">
-              Home
-            </a>
-            <span className="vads-u-color--black" key="dashboard">
-              <strong>My VA</strong>
-            </span>
-          </Breadcrumbs>
-
-          <h1 id="dashboard-title" data-testid="dashboard-title" tabIndex="-1">
-            My VA
-          </h1>
-
-          {props.showValidateIdentityAlert && (
-            <div className="vads-l-row">
-              <div className="vads-l-col--12 medium-screen:vads-l-col--8">
-                <IdentityNotVerified alertHeadline="Verify your identity to access more VA.gov tools and features" />
-              </div>
+  const renderDowntimeModal = (downtime, children) => {
+    if (downtime.status === externalServiceStatus.downtimeApproaching) {
+      return (
+        <>
+          <Modal
```

Comment:
Could this modal be its own component? We can keep it here for now, but I think it would be nice to keep the Dashboard component a bit lean

Reply:
Ideally, yes. Could be a good exercise. It would be a little tricky and maybe messy since this render method relies on `modalDismissed` and `setModalDismissed` which exist in this component's scope. 🤔